src/dataset-builder/CohortEditor.ts

```diff
@@ -151,27 +153,37 @@ export const CriteriaView = ({ criteria, deleteCriteria, updateCriteria }: Crite
   );
 
 let criteriaCount = 1;
-const createDefaultListCriteria = (listOption: ProgramDataListOption): ProgramDataListCriteria => {
-  return {
-    kind: 'list',
-    listOption,
-    name: listOption.name,
-    id: Number(_.uniqueId('')),
-    count: 100,
-    values: [listOption.values[0]],
-  };
-};
-
-const createDefaultRangeCriteria = (rangeOption: ProgramDataRangeOption): ProgramDataRangeCriteria => {
-  return {
-    kind: 'range',
-    rangeOption,
-    name: rangeOption.name,
-    id: criteriaCount++,
-    count: 100,
-    low: rangeOption.min,
-    high: rangeOption.max,
-  };
+const createDefaultProgramDataCriteria = async (
+  datasetId: string,
+  option: SnapshotBuilderProgramDataOption
+): Promise<ProgramDataRangeCriteria | ProgramDataListCriteria> => {
+  const generatedOptions = await DatasetBuilder().getProgramDataStatistics(datasetId, option);
+  switch (generatedOptions.kind) {
+    case 'range':
+      const rangeOption = generatedOptions as ProgramDataRangeOption;
+      return {
+        kind: 'range',
+        rangeOption,
+        name: rangeOption.name,
+        id: criteriaCount++,
```

Comment:
This is probably bogus and should changed; if a. unique ID is needed here we could generate it using `uniqueId()`.

Reply:
`_.uniqueId()` returns a string rather than a number. I agree that we probably don't need `id` anymore, but I think we are currently matching API definitions, so we will need to make API changes. I think the way to do that is to open a tech debt ticket to remove the unused fields.